src/sentry/testutils/modelmanifest.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from contextlib import ExitStack, contextmanager
+from dataclasses import dataclass
+from typing import Dict, Generator, Set, Type
+
+import django.apps
+
+from sentry.db.models import BaseManager, Model
+from sentry.db.models.manager.base import ModelManagerTriggerCondition
+from sentry.utils import json
+
+
+class ModelManifest:
+    """For auditing which models are touched by each test case."""
+
+    @dataclass(frozen=True, eq=True)
+    class Hit:
+        model_class: Type[Model]
+        condition: ModelManagerTriggerCondition
+
+        def as_json_output(self) -> Dict[str, str]:
+            return {
+                "model": self.model_class.__qualname__,
+                "condition": self.condition.name,
+            }
+
+    def __init__(self) -> None:
+        self.hits: Dict[str, Set[ModelManifest.Hit]] = defaultdict(set)
+
+    @contextmanager
+    def open(self, file_path: str) -> Generator[None, None, None]:
+        yield  # Populate self.hits
+
+        output = {
```

Comment:
I have one more request: another change I made on my branch (but reverted), as making this logic _first check for the existence of a previous manifest file at the given path_, and *merge any new test cases in*.

The reason is that, as this is installed as a pytest fixture, it runs on every test run, including filtered test runs.  If we simply replace the manifest file, it will get overriden with subsets of all tests when running tests with filters.  Having it merge instead makes the experience more simple.

Alternatively, we could reach into pytest internals and only output the new manifest if we confirm the entire suite is being run?  But I think I prefer just merging the current output with whatever had existed before.